From 4f90b768900c45da50d82c4f2f6cbd8e54d176f5 Mon Sep 17 00:00:00 2001
From: Jason Wang <jasowang@redhat.com>
Date: Mon, 29 Sep 2014 00:59:08 -0400
Subject: [virt] hyperv/net: Add support for netvsc build without CONFIG_SYSFS flag

Message-id: <1411952358-34104-25-git-send-email-jasowang@redhat.com>
Patchwork-id: 96769
O-Subject: [RHEL7.1 PATCH 24/34] Add support for netvsc build without CONFIG_SYSFS flag
Bugzilla: 1146357
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>

From: Haiyang Zhang <haiyangz@microsoft.com>

Bugzilla: 1146357

This change ensures the driver can be built successfully without the
CONFIG_SYSFS flag.
MS-TFS: 182270

Signed-off-by: Haiyang Zhang <haiyangz@microsoft.com>
Reviewed-by: K. Y. Srinivasan <kys@microsoft.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit e565e803d437b36c4fb4ced5e346827981183284)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/hyperv/netvsc_drv.c b/drivers/net/hyperv/netvsc_drv.c
index 6b737ef..6478bd3 100644
--- a/drivers/net/hyperv/netvsc_drv.c
+++ b/drivers/net/hyperv/netvsc_drv.c
@@ -644,8 +644,7 @@ int netvsc_recv_callback(struct hv_device *device_obj,
            packet->vlan_tci);
 
  skb_record_rx_queue(skb, packet->channel->
-       offermsg.offer.sub_channel_index %
-       net->real_num_rx_queues);
+       offermsg.offer.sub_channel_index);
 
  net->stats.rx_packets++;
  net->stats.rx_bytes += packet->total_data_buflen;
@@ -828,8 +827,6 @@ static int netvsc_probe(struct hv_device *dev,
  nvdev = hv_get_drvdata(dev);
  netif_set_real_num_tx_queues(net, nvdev->num_chn);
  netif_set_real_num_rx_queues(net, nvdev->num_chn);
- dev_info(&dev->device, "real num tx,rx queues:%u, %u\n",
-   net->real_num_tx_queues, net->real_num_rx_queues);
 
  ret = register_netdev(net);
  if (ret != 0) {
-- 
1.7.1